### PR TITLE
Replaced deprecated np.float with np.float64 in test_issue_14706

### LINKED ISSUE #21912: Deprecated `np.float` used in `test_sympify.py`
The test function here uses e.g. `np.float` rather than `np.float64`:
https://github.com/sympy/sympy/blob/81f1401300956b27e5dff870c37022421c7e37bb/sympy/core/tests/test_sympify.py#L721-L740
That gives warnings with recent numpy releases:
```python
>>> import numpy as np
>>> np.float
<stdin>:1: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
<class 'float'>
```
Those should be changed to e.g. `numpy.float64`.

### PR DESCRIPTION
**Pull request by ayushk7102**
#### References to other Issues or PRs
"Fixes #21912"

#### Brief description of what is fixed or changed
Replaced deprecated np.float used in test_sympify.py to np.float64

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Also `np.int` and `np.complex` need to be changed. You can see the warnings if you have a recent version of numpy and run the tests under pytest:
```console
$ pytest sympy/core/tests/test_sympify.py 
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.8.5, pytest-6.2.0, py-1.10.0, pluggy-0.13.1
architecture: 64-bit
cache:        yes
ground types: gmpy 2.0.8

rootdir: /Users/enojb/current/sympy/sympy, configfile: pytest.ini
plugins: instafail-0.4.2, xdist-2.1.0, cov-2.10.1, profiling-1.7.0, forked-1.3.0, doctestplus-0.9.0
collected 51 items                                                                                                                                           

sympy/core/tests/test_sympify.py ...........x...............................x.......                                                                   [100%]

====================================================================== warnings summary ======================================================================
sympy/core/tests/test_sympify.py::test_issue_14706
  /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:738: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    numpy.int(0),

sympy/core/tests/test_sympify.py::test_issue_14706
  /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:740: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    numpy.complex(0)]:

sympy/core/tests/test_sympify.py::test_issue_14706
  /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:758: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    numpy.int(1),

sympy/core/tests/test_sympify.py::test_issue_14706
  /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:760: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    numpy.complex(1)]:

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================= 49 passed, 2 xfailed, 4 warnings in 1.14s ==========================================================
```
Further changes can be committed to the same branch and pushed which will automatically update the PR (it isn't necessary to close the PR and open a new one).

**Comment by ayushk7102** [comment]
> white_check_mark
> 
> Hi, I am the [SymPy bot](https://github.com/sympy/sympy-bot) (v161). I'm here to help you write a release notes entry. Please read the [guide on how to write release notes](https://github.com/sympy/sympy/wiki/Writing-Release-Notes).
> 
>     * No release notes entry will be added for this pull request.
> 
> 
> Click here to see the pull request description that was parsed.

Hi, I have committed the required changes (np.int and np.complex) as well

**Comment by ayushk7102** [comment]
> Also `np.int` and `np.complex` need to be changed. You can see the warnings if you have a recent version of numpy and run the tests under pytest:
> 
> ```
> $ pytest sympy/core/tests/test_sympify.py 
> ==================================================================== test session starts =====================================================================
> platform darwin -- Python 3.8.5, pytest-6.2.0, py-1.10.0, pluggy-0.13.1
> architecture: 64-bit
> cache:        yes
> ground types: gmpy 2.0.8
> 
> rootdir: /Users/enojb/current/sympy/sympy, configfile: pytest.ini
> plugins: instafail-0.4.2, xdist-2.1.0, cov-2.10.1, profiling-1.7.0, forked-1.3.0, doctestplus-0.9.0
> collected 51 items                                                                                                                                           
> 
> sympy/core/tests/test_sympify.py ...........x...............................x.......                                                                   [100%]
> 
> ====================================================================== warnings summary ======================================================================
> sympy/core/tests/test_sympify.py::test_issue_14706
>   /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:738: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
>   Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
>     numpy.int(0),
> 
> sympy/core/tests/test_sympify.py::test_issue_14706
>   /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:740: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
>   Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
>     numpy.complex(0)]:
> 
> sympy/core/tests/test_sympify.py::test_issue_14706
>   /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:758: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
>   Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
>     numpy.int(1),
> 
> sympy/core/tests/test_sympify.py::test_issue_14706
>   /Users/enojb/current/sympy/sympy/sympy/core/tests/test_sympify.py:760: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
>   Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
>     numpy.complex(1)]:
> 
> -- Docs: https://docs.pytest.org/en/stable/warnings.html
> ========================================================= 49 passed, 2 xfailed, 4 warnings in 1.14s ==========================================================
> ```
> 
> Further changes can be committed to the same branch and pushed which will automatically update the PR (it isn't necessary to close the PR and open a new one).

Hi, I have committed and pushed the required changes (np.int and np.complex) as well

**Comment by oscarbenjamin** [comment]
> Hi, I have committed and pushed the required changes (np.int and np.complex) as well

It doesn't look like those changes have been pushed. Look at the "files changed" tab of the PR on GitHub to see the pushed changes.

**Comment by ayushk7102** [comment]
> > Hi, I have committed and pushed the required changes (np.int and np.complex) as well
> 
> It doesn't look like those changes have been pushed. Look at the "files changed" tab of the PR on GitHub to see the pushed changes.

Oh, my bad. Just fixed it

File: sympy/core/tests/test_sympify.py [hunk 1]
```diff
@@ -722,22 +722,22 @@ def test_issue_14706():
...
     assert numpy.all(x + z2 == numpy.full((2, 2), x))
     assert numpy.all(z1 + x == numpy.full((1, 1), x))
     assert numpy.all(z2 + x == numpy.full((2, 2), x))
     for z in [z3,
-              numpy.int(0),
-              numpy.float(0),
-              numpy.complex(0)]:
+              int(0),
+              numpy.float64(0),
+              complex(0)]:
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_sympify.py:740]: This test is supposed to test for numpy types so it should be e.g. `numpy.int64` rather than `int`.

Comment (ayushk7102) [reply sympy/core/tests/test_sympify.py]: Hi, I have pushed the necessary changes but my PR can't seem to get past the code-quality check (I get the message 'make sure all commits have an associated author'). I didn't have this error previously, could you please advise a solution?

**Comment by oscarbenjamin** [comment]
> my PR can't seem to get past the code-quality check (I get the message 'make sure all commits have an associated author').

See here:
https://groups.google.com/g/sympy/c/3DXOKgoeX24

The failure message from CI is:
```
The following authors were added to AUTHORS. If mailmap_update.py has
already been run and each author appears as desired and is not a
duplicate of some other author, then the changes can be committed.
Otherwise, see .mailmap for instructions on how to change an author's
entry.

	Ayush Kumar <65803868+ayushk7102@users.noreply.github.com>
	Ayush Kumar <ayushk7102@gmail.com>
```
The problem is that you have made commits with different email addresses. I think that the noreply email address is used by GitHub if you make commits using the web UI.

There are two options to solve this:

1. Squash all of the commits into a single commit using the primary email address.
2. Add a line to `.mailmap` specifying which name/email should be used in the author information.

If you have done that and `bin/mailmap_update.py` no longer shows any output then you can run `bin/update_authors.py` which will add your name to the authors file. You need to commit and push any changes to the authors file or the .mailmap file.

**Comment by oscarbenjamin** [comment]
You might find it easier just to make a new PR from a new branch.

**Comment by ayushk7102** [comment]
> You might find it easier just to make a new PR from a new branch.

Right, I'll try that. Thanks a lot for the help.

**Comment by oscarbenjamin** [comment]
> > You might find it easier just to make a new PR from a new branch.
> 
> Right, I'll try that. Thanks a lot for the help.

Make sure to start from the most up to date master branch.

**Pull request closed without merging**